app/src/org/commcare/heartbeat/HeartbeatRequester.java

```diff
@@ -96,10 +98,20 @@ public void parseResponse(JSONObject responseAsJson) {
             attemptApkUpdateParse(responseAsJson);
             attemptCczUpdateParse(responseAsJson);
             checkForForceLogs(responseAsJson);
+            checkForDisableBackgroundWork(responseAsJson);
         }
         DriftHelper.clearMaxDriftSinceLastHeartbeat();
     }
 
+    private void checkForDisableBackgroundWork(JSONObject responseAsJson) {
+        boolean disableBackgroundWork = responseAsJson.optBoolean("disable_background_work", false);
+        HiddenPreferences.setDisableBackgroundWork(disableBackgroundWork);
```

Comment:
Right, I think it makes sense then to enable these on next login in that case and then disable again if the flag in heartbeat still conveys that we should stop ?

Reply:
@ctsims  bump

Reply:
I'm not sure how many phones successfully get a heartbeat every day (and when in the day they get it if they do, some CAS users get their heartbeat in the afternoon), so it might be worth waiting slightly longer than a day, but the pattern you describe sounds right.

Reply:
@ctsims [Changed the behaviour](https://github.com/dimagi/commcare-android/pull/2195/commits/32aaf0774baec7fcb676ceaef1b22c96dbd48ed0) for work to be disabled for 36 hours since the phone encounters the heartbeat flag to disable background work.